cmd/svcat/class/get_cmd_test.go

```diff
@@ -174,3 +177,126 @@ func TestListClasses(t *testing.T) {
 		})
 	}
 }
+
+var _ = Describe("Get Classes Command", func() {
+	Describe("NewGetClassesCmd", func() {
+		It("Builds and returns a cobra command", func() {
+			cxt := &command.Context{}
+			cmd := NewGetCmd(cxt)
+			Expect(*cmd).NotTo(BeNil())
+			Expect(cmd.Use).To(Equal("classes [NAME]"))
+			Expect(cmd.Short).To(ContainSubstring("List classes, optionally filtered by name, scope or namespace"))
+			Expect(cmd.Example).To(ContainSubstring("svcat get classes"))
+			Expect(cmd.Example).To(ContainSubstring("svcat get classes --scope cluster"))
+			Expect(cmd.Example).To(ContainSubstring("svcat get classes --scope namespace --namespace dev"))
+			Expect(len(cmd.Aliases)).To(Equal(2))
+		})
+	})
+	Describe("Validate", func() {
+		It("allows class name arg to be empty", func() {
+			cmd := &getCmd{}
+			err := cmd.Validate([]string{})
+			Expect(err).To(BeNil())
+		})
+		It("optionally parses the class name argument", func() {
+			cmd := &getCmd{}
+			err := cmd.Validate([]string{"mysqldb"})
+			Expect(err).To(BeNil())
+			Expect(cmd.name).To(Equal("mysqldb"))
+		})
+	})
+	Describe("getAll", func() {
```

Comment:
generally you don't have to test private methods, as they should be subsumed in whatever public methods use them. I would change this to test Run(), and control the different cases by modifying whether you pass in a uuid or name.

Reply:
it's not the testing a single function, it's that it's a private function - if the content of getAll was just inlined in Run, it wouldn't make sense to try to test the functionality of that block specifically, and because it's private, it kinda is just inlined.

This wouldn't be a big change, just switch the describe to "Run", and change the calls to to getAll with calls to Run with the appropriate fields filled in.

Reply:
Add a comment to the test about not worrying about it if it breaks. I'm not hung up on idealogical test purity that says you only test public methods.

Reply:
Why would we add a test that we don't care if it breaks? And you do test private methods, but implicitly. A test for Run(), which I think we should definitely have, would be 99% the same lines of code as the tests she's written here for getAll() and a hypothetical test for get()